Snowflake.Data.Tests/TestDataGenarator.cs

```diff
@@ -3,63 +3,86 @@
         
         public static bool NextBool()
         {
-            return random.Next(0, 1) == 1;
+            return s_random.Next(0, 1) == 1;
         }
 
         public static int NextInt(int minValueInclusive, int maxValueExclusive)
         {
-            return random.Next(minValueInclusive, maxValueExclusive);
+            return s_random.Next(minValueInclusive, maxValueExclusive);
         }
         
         public static string NextAlphaNumeric()
         {
-            return NextLetter() +
-                   Enumerable.Repeat(alphanumericChars, random.Next(5, 12))
-                       .Select(NextChar)
-                       .Aggregate((s1, s2) => s1 + s2);
+            return NextAlphaNumeric(s_random.Next(5, 12));
         }
 
-        public static string NextDigitsString(int length)
+        public static string NextAlphaNumeric(int length)
         {
-            if (length == 1)
+            if (length < 1)
+            {
+                return "";
+            }
+            var buffer = new char[length];
+            buffer[0] = NextLetterChar();
+            for (var i = 1; i < length; i++)
             {
-                return NextNonZeroDigitString();
-            } 
-            return NextNonZeroDigitString() + Enumerable.Repeat(digitChars, length - 1)
-                .Select(NextChar)
-                .Aggregate((s1, s2) => s1 + s2);
+                buffer[i] = NextAlphaNumericChar();
+            }
+            return new string(buffer);
         }
 
-        public static string NextNonZeroDigitString()
+        public static string NextDigitsString(int length)
         {
-            return NextChar(nonZeroDigits);
-        }
+            if (length < 1)
+            {
+                return "";
+            }
 
-        private static string NextLetter()
-        {
-            return NextChar(letterChars);
+            if (length == 1)
+            {
+                NextDigitAsString();
+            }
+            var buffer = new char[length];
+            buffer[0] = NextNonZeroDigitChar();
+            for (var i = 1; i < length; i++)
+            {
+                buffer[i] = NextDigitChar();
+            }
+            return new string(buffer);
         }
+        
+        private static char NextAlphaNumericChar() => NextChar(s_alphanumericChars);
+        
+        public static string NextNonZeroDigitAsString() => NextNonZeroDigitChar().ToString();
 
-        private static string NextChar(string chars)
-        {
-            return chars[random.Next(chars.Length)].ToString();
-        }
+        private static char NextNonZeroDigitChar() => NextChar(s_nonZeroDigits);
+        
+        private static string NextDigitAsString() => NextDigitChar().ToString(); 
+        
+        private static char NextDigitChar() => NextChar(s_digitChars);
+        
+        private static string NextLetterAsString() => NextLetterChar().ToString();
```

Comment:
Why add functions that aren't used? It reduced code coverage.

Reply:
I look at this class as like a library toolbox. Sooner or later they will be used. As I understand code coverage is related to code in the "production code". And this is just a testing helper class which should not be measured in code coverage.